core/src/main/java/com/mystudio/gamename/Game.java

```
@@ -0,0 +1,211 @@
+                if (!bullets[i].getCollisionBox().intersects(asteroid.getCollisionCircle())) {
+                    continue;
+                }
+                bullets[i].setActive(false);
+                myGameSounds.explode.play();
+                splitAsteroid(asteroid);
+                break;
+            }
+        } // end for loop
+    } // end checkBulletAsteroidCollisions
+
+    private void checkAsteroidPlayerCollisions() {
+        for (Asteroid asteroid : activeAsteroids) {
+            if (!asteroid.getCollisionCircle().intersects(player.getPlayerCollisionCircle())) {
+                continue;
+            }
+            player.setActive(false);
+            lives.removeLife();
+            myGameSounds.death.play();
+            ui.set(lives.getRemainingLives(),score.getScore());
+        }
+    }
+
+    private void splitAsteroid(Asteroid asteroid) {
+        if (asteroid.getSize() == Asteroid.SIZE_LARGE_ASTEROID) {
+            placeAsteroid(MAX_LARGE, MAX_MEDIUM + MAX_LARGE,
+                    asteroid.getCollisionCircle().getX(),
+                    asteroid.getCollisionCircle().getY()
+            );
+            score.changeScoreByAmount(10);
+        } else if (asteroid.getSize() == Asteroid.SIZE_MEDIUM_ASTEROID) {
+            placeAsteroid(MAX_MEDIUM + MAX_LARGE,
+                    MAX_SMALL + MAX_MEDIUM + MAX_LARGE,
+                    asteroid.getCollisionCircle().getX(),
+                    asteroid.getCollisionCircle().getY()
+            );
+            score.changeScoreByAmount(50);
+        } else {
+            score.changeScoreByAmount(200);
+        }
+        ui.set(lives.getRemainingLives(),score.getScore());
+        asteroid.setActive(false);
+        activeAsteroids.remove(asteroid);
+    }
+
+    public void placeAsteroid(int from, int to, float posX, float posY) {
+        int numberOfAsteroidsSpawned = 0;
+        for (int i = from; i < to; i++) {
+            if (!asteroids[i].isActive()) {
```

This should change to fail fast.